delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/ElasticSearchViews.scala

```diff
@@ -0,0 +1,130 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch
+
+import cats.effect.Clock
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewCommand._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewEvent._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewRejection._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewState.{Current, Initial}
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewValue._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model._
+import ch.epfl.bluebrain.nexus.delta.sdk.Permissions
+import ch.epfl.bluebrain.nexus.delta.sdk.utils.IOUtils
+import monix.bio.{IO, UIO}
+
+class ElasticSearchViews {}
+
+object ElasticSearchViews {
+
+  /**
+    * The elasticsearch module type.
+    */
+  final val moduleType: String = "elasticsearchview"
+
+  private[elasticsearch] def next(
+      state: ElasticSearchViewState,
+      event: ElasticSearchViewEvent
+  ): ElasticSearchViewState = {
+    // format: off
+    def created(e: ElasticSearchViewCreated): ElasticSearchViewState = state match {
+      case Initial     => Current(e.id, e.project, e.value, e.source, Map.empty, e.rev, deprecated = false,  e.instant, e.subject, e.instant, e.subject)
+      case s: Current  => s
+    }
+
+    def updated(e: ElasticSearchViewUpdated): ElasticSearchViewState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, value = e.value, source = e.source, updatedAt = e.instant, updatedBy = e.subject)
+    }
+
+    def tagAdded(e: ElasticSearchViewTagAdded): ElasticSearchViewState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, tags = s.tags + (e.tag -> e.targetRev), updatedAt = e.instant, updatedBy = e.subject)
+    }
+    // format: on
+
+    def deprecated(e: ElasticSearchViewDeprecated): ElasticSearchViewState = state match {
+      case Initial    => Initial
+      case s: Current => s.copy(rev = e.rev, deprecated = true, updatedAt = e.instant, updatedBy = e.subject)
+    }
+
+    event match {
+      case e: ElasticSearchViewCreated    => created(e)
+      case e: ElasticSearchViewUpdated    => updated(e)
+      case e: ElasticSearchViewTagAdded   => tagAdded(e)
+      case e: ElasticSearchViewDeprecated => deprecated(e)
+    }
+  }
+
+  private[elasticsearch] def evaluate(permissions: UIO[Permissions])(
```

Comment:
I don't think we need the UIO here. I got rid of it in the previous PR on Acls state machine